Bottom: 9cc5d8faa6dffbecba016e3a93cb67ed60faa95d
Top:    f6aaea9b20464e86e82b553abd6a5cb2ff372dd8
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-18 17:14:14 +0800

Refresh of vhost_net-batch-update-used

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 164ba4c..90d1035 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -451,8 +451,7 @@ static int vhost_net_tx_get_vq_desc(struct vhost_net *net,
 				  out_num, in_num, NULL, NULL);
 
 	if (r == vq->num && vq->busyloop_timeout) {
-		/* FLush batched heads first */
-		if (!nvq->ubufs)
+		if (vhost_sock_zcopy(vq->private_data))
 			vhost_net_signal_used(nvq);
 		preempt_disable();
 		endtime = busy_clock() + vq->busyloop_timeout;
